src/Domain/CoordinateMaps/EquatorialCompression.hpp

```diff
@@ -26,20 +26,24 @@ namespace CoordinateMaps {
  * \brief Redistributes gridpoints on the sphere.
  * \image html EquatorialCompression.png "A sphere with an `aspect_ratio` of 3."
  *
- * \details A mapping from the sphere to itself which depends on a single
- * parameter, the `aspect_ratio` \f$\alpha\f$, which is the ratio of the
- * horizontal length to the vertical height for a given point. This parameter
- * name was chosen because points with \f$\tan \theta = 1\f$ get mapped to
- * points with \f$\tan \theta' = \alpha\f$. In general, gridpoints located
- * at an angle \f$\theta\f$ from the pole are mapped to a new angle
+ * \details A mapping from the sphere to itself which redistributes points
+ * towards a user-specifed axis, indicated by `index_pole_axis_`.
+ * Once the axis is selected, the map is determined by a single parameter,
+ * the `aspect_ratio` \f$\alpha\f$, which is the ratio of the distance
+ * perpendicular to the polar axis to the distance along the polar axis for
+ * a given point. This parameter name was chosen because points with
+ * \f$\tan \theta = 1\f$ get mapped to points with \f$\tan \theta' = \alpha\f$.
+ * In general, gridpoints located at an angle \f$\theta\f$ from the pole are
+ * mapped to a new angle
```

Comment:
Question: Do I understand correctly that this map only allowed the compression in one angular dimension of a sphere before, but now you want to apply it to both angular dimensions? So, will you apply this map twice? How is this change related to the half-wedges that you mention in the PR description?

Reply:
The angle theta that is referenced here is the pole angle (the z axis in the current code), the azimuthal angle of mapped coordinates is left unchanged.

This PR allows the pole axis to be changed to a different axis, and the pole angle of a mapped coordinate will change with respect to that axis (the azimuthal angle of a mapped coordinate with respect to that axis is left unchanged).

Another way to think about the mapping being done is that if the existing map is EC(x), the new map is R^{-1}EC(R(x)), where R is a rotation matrix that rotates the z axis to the user specified axis.

The Half Wedges in spherical domains, including the BinaryCompactObject domain, are obtained by splitting ordinary Wedges in half down the y-z plane, which leads to a compressed grid close to the y-z plane. This can be undone by applying an EC map with the pole axis set to be the x axis, so that the equator is the y-z plane. The current EC map's pole axis is hard coded to be the z axis. With this addition the two options of Half Wedges and EC can be used together to obtain a spherical shell of 10 (Half) Wedge blocks where each (Half) Wedge has the same angular size on the sphere.

Reply:
Ok I think I understand now, thanks for explaining! We take the x=0 plane as the equatorial plane and re-distribute grid points _away_ from it, to make the half-wedges larger.

What I don't yet understand: When the original map was equiangular, then the equatorially compressed (or elongated) map will not be equiangular anymore, or will it? Is it feasible to restore the equiangular property? Intuitively it seems to me that we don't want to continuously distort the angular dimension so the density of points near the equator is larger than near the poles (which I think the equatorial elongation does). Instead, I think we would like to keep the density of points constant along the polar angle ("equiangular") and divide up the polar dimension into equally-sized blocks. Specifically, split the outer shell at polar angle 30deg, at 90deg (the equator, i.e. the x=0 plane), and at 150deg. For the `Wedge` map this seems possible by specifying the opening angle in both angular dimensions. For `Frustum` I don't yet understand the map well enough to know if it's feasible to begin a radial grid line at a corner of the inner cube and end it at polar angle 30 / 90 / 150, instead of the current 45 / 90 / 135. Does what I'm thinking out loud here make sense at all? Is it perhaps already solved by the equatorial compression map, or can it, or do we need to build this into `Wedge` and `Frustum`? Please let me know what you think.

Reply:
The equiangular vs equidistant distinction can be boiled down to the computation domain(CD) metric term `g_{\xi \xi}` to see how changes in the logical coordinate relate to distances in the physical domain. Wanting this term to be constant (indicating logical distances map linearly to physical distances along a (curved) surface) determines whether equidistant or equiangular coordinates should be used. Check out [redistributing gridpoints](https://spectre-code.org/redistributing_gridpoints.html) to see how the equiangular distribution can be solved for.

Using an Equatorial Compression (EC) map on a map this is equiangular (EA) to begin with (constant computational domain metric along the spherical surface) will lead to a non-constant CD metric. But whether it is still a "good" or "better than equidistant" map is determined by the behavior of the CD metric, perhaps not all non-constant CD metrics are "just as bad" as using an equidistant(ED) map. (I still need to look into this, but my intuition is that an EC+EA map is still better than an ED map, even if EC+EA != EA.)

I think an EC+EA map might lead to a distribution of points which is "locally EA" which would still be better than ED.
I think there is a value of `aspect_ratio` for the EC such that the Half Wedges have the same angular size as the full Wedges - such that a Shell of 2 Wedges + 8 Half Wedges will have 10 Blocks all of the same angular size on the sphere, with a gridpoint distribution that reflects that. If I understand correctly I think this is (close to) what you want? I am still thinking about your second paragraph but let me know if what I'm saying helps you narrow down what we might want for the domain!